### PR TITLE
fix #19571 by removing dataprotection.azure packages

### LINKED ISSUE #19571: Remove DataProtection.AzureKeyVault and DataProtection.AzureStorage from 5.0 build
The Azure SDK team is taking over these packages: https://github.com/aspnet/Announcements/issues/408

As a result, during 5.0 we should remove these packages and post a breaking change announcement indicating that users should migrate to the new packages.

### PR DESCRIPTION
**Pull request by analogrelay**
Closes #19571

I'm pretty sure I probably missed something so we'll see what the CI does. As per https://github.com/aspnet/Announcements/issues/408 we will be removing these packages in 5.0 since the Azure SDK team is taking them over (with new package IDs). This PR removes the DataProtection.Azure packages **and** pre-emptively removes references to Microsoft.Extensions.Configuration.AzureKeyVault (which will be removed in https://github.com/dotnet/extensions/issues/3057).

I *probably missed something* but the CI will be more effective at finding it that I am :).

TODO before merging:
* [x] Ensure packages no longer show up in build artifacts.

### CODE REVIEW

**Review by Pilchie — APPROVED** [review]
Same question about 5.0 Preview 1.

File: eng/Dependencies.props [hunk 1]
```diff
@@ -161,10 +161,6 @@ and are generated based on the last package release.
     <LatestPackageReference Include="IdentityServer4.Storage" Version="$(IdentityServer4StoragePackageVersion)" />
     <LatestPackageReference Include="Libuv" Version="$(LibuvPackageVersion)" />
     <LatestPackageReference Include="MessagePack" Version="$(MessagePackPackageVersion)" />
-    <LatestPackageReference Include="Microsoft.Azure.KeyVault" Version="$(MicrosoftAzureKeyVaultPackageVersion)" />
```
Comment (BrennanConroy) [inline eng/Dependencies.props]: Can we remove these from `Versions.props` as well?

Comment (anurse) [reply eng/Dependencies.props]: Probably. I just missed it while grepping around :). I'll do that.

File: src/DataProtection/AzureKeyVault/src/Microsoft.AspNetCore.DataProtection.AzureKeyVault.csproj [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <PackageTags>aspnetcore;dataprotection;azure;keyvault</PackageTags>
-    <IsPackable>true</IsPackable>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <Reference Include="Microsoft.AspNetCore.DataProtection" />
-    <Reference Include="Microsoft.Azure.KeyVault" />
-    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" />
```
Comment (dougbu) [inline src/DataProtection/AzureKeyVault/src/Microsoft.AspNetCore.DataProtection.AzureKeyVault.csproj]: Do we still need the Microsoft.IdentityModel.Clients.ActiveDirectory dependency e.g. in eng/Versions.props? Please remove if possible.

**Comment by anurse** [comment]
Rebaséd

File: src/DataProtection/AzureStorage/src/Microsoft.AspNetCore.DataProtection.AzureStorage.csproj [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <GenerateDocumentationFile>true</GenerateDocumentationFile>
-    <PackageTags>aspnetcore;dataprotection;azure;blob</PackageTags>
-    <IsPackable>true</IsPackable>
-    <UseLatestPackageReferences>true</UseLatestPackageReferences>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <Reference Include="Microsoft.AspNetCore.DataProtection" />
-    <Reference Include="Microsoft.Azure.Storage.Blob" />
```
Comment (dougbu) [inline src/DataProtection/AzureStorage/src/Microsoft.AspNetCore.DataProtection.AzureStorage.csproj]: Do we still need the Microsoft.Azure.Storage.Blob dependency e.g. in eng/Versions.props?

**Review by dougbu — APPROVED** [review]

File: src/DataProtection/AzureKeyVault/src/Microsoft.AspNetCore.DataProtection.AzureKeyVault.csproj (hunk 2 above)
Comment (anurse) [reply src/DataProtection/AzureKeyVault/src/Microsoft.AspNetCore.DataProtection.AzureKeyVault.csproj]: We use it in the Auth middleware.

File: src/DataProtection/AzureStorage/src/Microsoft.AspNetCore.DataProtection.AzureStorage.csproj (hunk 3 above)
Comment (anurse) [reply src/DataProtection/AzureStorage/src/Microsoft.AspNetCore.DataProtection.AzureStorage.csproj]: Yep, I already removed it.

**Comment by anurse** [comment]
I think we're ready to go here. I'll rebase and resolve conflicts and if we get a green build I plan to merge.

**Comment by anurse** [comment]
Ported to preview2 and now have to rebase and respin builds :(.

**Comment by anurse** [comment]
Had to do battle with merge conflicts on the version files. I think I can claim victory now. There should be no **changes** to versions in this PR, just removed dependencies.

**Comment by anurse** [comment]
Windows test build hanged?

**Comment by anurse** [comment]
IT PASSED?!

**Pull request merged**